libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2818,6 +2831,13 @@ bool GCS_MAVLINK::try_send_message(const enum ap_message id)
         send_sensor_offsets();
         break;
 
+    case MSG_SIMSTATE:
+        CHECK_PAYLOAD_SIZE(SIMSTATE);
+        send_simstate();
```

Comment:
I'd like to see the CHECK_PAYLOAD_SIZE changed to be inside the send_simstate method: that way, in non-SITL (or if sitl is nullptr), it won't re-queue this to be sent just because it can't send SIMSTATE (although it would never send it).

Reply:
That would mean making it `bool` and generally messy.  It would also not match the code around it.  It can do with help, but later.

Reply:
Why would it be messy? Other cases in the same switch already check the return value of a method...

By the way, this was a change for Copter and Sub that previously excluded this code when it was a SITL build 😛